### PR TITLE
[TE] rootcause - max-score entity set

### PR DESCRIPTION
**Pull request by apucher**
* max score set for entities to simplify conflict resolution between multiple instances of the same entity within a pipeline

* use max-score set to implement anomaly and holiday pipelines

* standardize pipelines to resolve conflicts via max score

* compound scoring strategy for holidays taking advantage of max dimension scores

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MaxScoreSet.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        it.remove();
+        changed = true;
+      }
+    }
+
+    return changed;
+  }
+
+  @Override
+  public boolean removeAll(Collection<?> c) {
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MaxScoreSet.java]: let this class extend AbstractSet.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java [hunk 2]
```diff
@@ -185,6 +172,23 @@ private static double makeDimensionScore(Map<String, DimensionEntity> urn2entity
       }
       return urns;
     }
+  }
+
+  private static class CompoundStrategy implements ScoringStrategy {
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: javadoc please !

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java [hunk 3]
```diff
@@ -32,7 +30,8 @@
     LINEAR,
     TRIANGULAR,
     QUADRATIC,
-    DIMENSION
+    DIMENSION,
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: code documentation about the strategies please !

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MaxScoreSet.java (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MaxScoreSet.java]: this uses a map as a delegate. AbstractSet would store items internally

**Pull request merged**